core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -57,6 +57,9 @@ object UnitAutomation {
                         )
                         && unit.movement.canMoveTo(it)
```

Comment:
I think the correct solution would be to add a `&& unit.movement.canReach(it)` here

That way you solve the 'double path' you have below, and as a bonus if the first viewable and movable ruin happens to not be reachable you can directly head towards the second one:
- First

Reply:
Done :+1: